tests/test_core/test_new_genetic_code.py

```diff
@@ -75,3 +81,16 @@ def test_sizeframes():
     }
     got = set(code.sixframes(seq))
     assert got == expect
+
+
+@pytest.mark.parametrize("code_id", range(1, 5))
+def test_compare_codes_true(code_id):
+    gc = new_genetic_code.get_code(code_id)
+    assert gc == gc
```

Comment:
something will always equal itself, so that must be an error. What are you trying to test? Perhaps you mean `gc.ID == code_id`?